integrations/mcp.md

```diff
@@ -0,0 +1,120 @@
+---
+layout: integration
+name: mcp
+description: Haystack Tool Integration with the Model Context Protocol (MCP)
+
+authors:
+    - name: deepset
+      socials:
+        github: deepset-ai
+        twitter: haystack_ai
+        linkedin: https://www.linkedin.com/company/deepset-ai/
+pypi: https://pypi.org/project/mcp-haystack/
+repo: https://github.com/deepset-ai/haystack-core-integrations/tree/main/integrations/mcp
+type: Tool Integration
+report_issue: https://github.com/deepset-ai/haystack-core-integrations/issues
+logo: /logos/mcp.png
+version: Haystack 2.0
+toc: true
+---
+### **Table of Contents**
+- [Overview](#overview)
+- [Installation](#installation)
+- [Usage](#usage)
+- [Examples](#examples)
+- [License](#license)
+
+## Overview
+
+MCP Haystack Integration adds support for the Model Context Protocol (MCP) to Haystack. MCP is an open protocol that standardizes how applications provide context to LLMs, similar to how USB-C provides a standardized way to connect devices.
+
+This integration allows you to easily connect external tools and services to your Haystack pipelines using the Model Context Protocol, enabling more powerful and flexible agentic applications.
+
+## Installation
+
+```bash
+pip install mcp-haystack
+```
+
+## Usage
+
+```python
+from haystack_integrations.components.tools.mcp import MCPTool, SSEServerInfo
+
+# Create an MCP tool that connects to an HTTP server
+server_info = SSEServerInfo(base_url="http://localhost:8000")
+tool = MCPTool(name="my_tool", server_info=server_info)
+
+# Use the tool
+result = tool.invoke(param1="value1", param2="value2")
+```
+
+## Examples
+
+Check out the examples directory to see practical demonstrations of how to integrate the MCPTool into Haystack's tooling architecture. These examples will help you get started quickly with your own agentic applications.
+
+### What is uvx?
+
+In some examples below, we use the `StdioServerInfo` class which relies on uvx behind the scenes. uvx is a convenient command from the uv package that runs Python tools in temporary, isolated environments. You only need to install uvx once, and it will automatically fetch any required packages on first use without needing manual installation.
+
+### Example 1: MCP Server with SSE Transport
+
+This example demonstrates how to create a simple calculator server using MCP and connect to it using the MCPTool with SSE transport.
+
+**Step 1: Run the MCP Server**
+
+First, run the server that exposes calculator functionality (addition and subtraction) via MCP:
```

Comment:
Let's add a link to the file in the GitHub repo so that users can easily inspect the file. Here it would be https://github.com/deepset-ai/haystack-core-integrations/blob/main/integrations/mcp/examples/mcp_sse_server.py